src/xdocs/config_filters.xml

```diff
@@ -2232,6 +2232,296 @@ public class UserService {
       </subsection>
     </section>
 
+    <section name="SuppressWithNearbyTextFilter">
+      <p>Since Checkstyle 10.3.4</p>
+      <subsection name="Description" id="SuppressWithNearbyTextFilter_Description">
+        <p>
+          Filter <code>SuppressWithNearbyTextFilter</code> uses plain text to suppress
+          nearby audit events. The filter can only suppress checks which have Checker as a
+          parent module and knows nothing about AST. It supports only singe-line comments.
+        </p>
+        <p>
+          Rationale: Sometimes there are legitimate reasons for violating a check.
+          When this is a matter of the code in question and not personal preference,
+          the best place to override the policy is in the code itself. Semi-structured
+          comments can be associated with the check. This is sometimes superior to
+          a separate suppressions file, which must be kept up-to-date as the source
+          file is edited.
+        </p>
+      </subsection>
+      <subsection name="Properties" id="SuppressWithNearbyTextFilter_Properties">
+        <div class="wrapper">
+          <table>
+            <tr>
+              <th>name</th>
+              <th>description</th>
+              <th>type</th>
+              <th>default value</th>
+              <th>since</th>
+            </tr>
+            <tr>
+              <td>nearbyTextPattern</td>
+              <td>Specify nearby text pattern to trigger filter to begin suppression.</td>
+              <td><a href="property_types.html#Pattern">Pattern</a></td>
+              <td><code>"SUPPRESS CHECKSTYLE (\w+)"</code></td>
+              <td>10.3.4</td>
+            </tr>
+            <tr>
+              <td>checkPattern</td>
+              <td>Specify check name pattern to suppress.</td>
+              <td><a href="property_types.html#Pattern">Pattern</a></td>
+              <td><code>".*"</code></td>
+              <td>10.3.4</td>
+            </tr>
+            <tr>
+              <td>messagePattern</td>
+              <td>Specify check violation message pattern to suppress.</td>
+              <td><a href="property_types.html#Pattern">Pattern</a></td>
+              <td><code>null</code></td>
+              <td>10.3.4</td>
+            </tr>
+            <tr>
+              <td>idPattern</td>
+              <td>Specify check ID pattern to suppress.</td>
+              <td><a href="property_types.html#Pattern">Pattern</a></td>
+              <td><code>null</code></td>
+              <td>10.3.4</td>
+            </tr>
+            <tr>
+              <td>influenceFormat</td>
+              <td>Specify negative/zero/positive value that defines the number of lines
+                  preceding/at/following the suppression comment.</td>
```

Comment:
what words we can use mention that it is also index in regexp?

Reply:
What do you mean? It is supposed to be a string that is later parsed to an integer, not a Regex.

Reply:
We need some hint that it can be `$x`, so it is not a number (`negative/zero/positive value`)

Reply:
Yes, please apply to checkPattern.
User need small hint, but in examples he will find how it works

Reply:
Property can also be an regexp group index and be picked from target file by nearbyTextPattern

Reply:
Done. Applied to both.